src/com/owncloud/android/services/observer/FileObserverService.java

```diff
@@ -276,16 +276,27 @@ private void addObservedFile(OCFile file, Account account) {
      */
     private void addObservedFile(String localPath, Account account) {
         File file = new File(localPath);
-        String parentPath = file.getParent();
-        FolderObserver observer = mFolderObserversMap.get(parentPath);
-        if (observer == null) {
-            observer = new FolderObserver(parentPath, account, getApplicationContext());
-            mFolderObserversMap.put(parentPath, observer);
-            Log_OC.d(TAG, "Observer added for parent folder " + parentPath + "/");
+
+        if(file.isDirectory()) {
```

Comment:
This will not work if no file of the folder was downloaded before, the folder will not exist in the local file system.